### ISSUE TITLE
ue4-docker build 4.22.1 fails on Windows, not enough space, but have plenty

### ISSUE THREAD

**Issue by ki11roy** [adamrehn/ue4-docker#37]
Output of the `ue4-docker info` command:
```
PS C:\Users\a_selivanov> ue4-docker info
ue4-docker version:         0.0.31 (latest available version is 0.0.32)
Operating system:           Windows 10 Enterprise Version 1809 (OS Build 17763.503)
Docker daemon version:      18.09.2
NVIDIA Docker supported:    No
Maximum image size:         1000GB
Available disk space:       1.33 TiB
Total system memory:        63.95 GiB physical, 73.45 GiB virtual
Number of processors:       6 physical, 12 logical
```

Output of the `docker info` command:
```
PS C:\Users\a_selivanov> ue4-docker info
ue4-docker version:         0.0.31 (latest available version is 0.0.32)
Operating system:           Windows 10 Enterprise Version 1809 (OS Build 17763.503)
Docker daemon version:      18.09.2
NVIDIA Docker supported:    No
Maximum image size:         1000GB
Available disk space:       1.33 TiB
Total system memory:        63.95 GiB physical, 73.45 GiB virtual
Number of processors:       6 physical, 12 logical
PS C:\Users\a_selivanov> docker info
Containers: 0
 Running: 0
 Paused: 0
 Stopped: 0
Images: 49
Server Version: 18.09.2
Storage Driver: windowsfilter
 Windows:
Logging Driver: json-file
Plugins:
 Volume: local
 Network: ics l2bridge l2tunnel nat null overlay transparent
 Log: awslogs etwlogs fluentd gelf json-file local logentries splunk syslog
Swarm: inactive
Default Isolation: hyperv
Kernel Version: 10.0 17763 (17763.1.amd64fre.rs5_release.180914-1434)
Operating System: Windows 10 Enterprise Version 1809 (OS Build 17763.503)
OSType: windows
Architecture: x86_64
CPUs: 12
Total Memory: 63.95GiB
Name: BY1-CABuilds-01
ID: ZAXJ:OC2L:MUXM:QHW3:J4SD:KVW7:KIO6:UY57:GEVO:73PA:J33L:ZQS2
Docker Root Dir: e:\docker_windows
Debug Mode (client): false
Debug Mode (server): true
 File Descriptors: -1
 Goroutines: 26
 System Time: 2019-05-31T14:29:17.4748918+03:00
 EventsListeners: 1
Registry: https://index.docker.io/v1/
Labels:
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false
Product License: Community Engine
```

Docker daemon settings
```
{
  "registry-mirrors": [],
  "insecure-registries": [],
  "debug": true,
  "log-level": "debug",
  "experimental": false,
  "group": "docker-users",
  "data-root": "e:\\docker_windows",
  "storage-opts": [
    "size=1000G"
  ]
}
```

```
PS D:\juyd> ue4-docker build 4.22.1 --no-engine
[ue4-docker build] WINDOWS CONTAINER SETTINGS
[ue4-docker build] Isolation mode:               process
[ue4-docker build] Base OS image tag:            ltsc2019 (host OS is Windows 10 version 1809)
[ue4-docker build] Memory limit:                 No limit
[ue4-docker build] Detected max image size:      1000GB
[ue4-docker build] Directory to copy DLLs from:  C:\Windows\System32

[ue4-docker build] GENERAL SETTINGS
[ue4-docker build] Not excluding any Engine components.

Retrieving the Git credentials that will be used to clone the UE4 repo
Username: xxxxx
Password:


[ue4-docker build] Image "adamrehn/ue4-build-prerequisites:ltsc2019" exists and rebuild not requested, skipping build.

[ue4-docker build] Image "adamrehn/ue4-source:4.22.1-ltsc2019" exists and rebuild not requested, skipping build.

[ue4-docker build] User specified `--no-engine`, skipping ue4-engine image build.

[ue4-docker build] Building image "adamrehn/ue4-minimal:4.22.1-ltsc2019"...
Sending build context to Docker daemon  11.78kB
Step 1/22 : ARG NAMESPACE
Step 2/22 : ARG TAG
Step 3/22 : ARG PREREQS_TAG
Step 4/22 : FROM ${NAMESPACE}/ue4-source:${TAG}-${PREREQS_TAG} AS builder
 ---> e37c1dde37fb
Step 5/22 : COPY set-changelist.py C:\set-changelist.py
 ---> Using cache
 ---> 1d2ad652ec17
Step 6/22 : RUN python C:\set-changelist.py C:\UnrealEngine\Engine\Build\Build.version && echo. && echo.RUN directive complete. Docker will now commit the filesystem layer to disk. && echo.Note that for large filesystem layers this can take quite some time. && echo.Performing filesystem layer commit... && echo.
 ---> Using cache
 ---> 97567c285356
Step 7/22 : COPY patch-filters-xml.py C:\patch-filters-xml.py
 ---> Using cache
 ---> 7bda3ec53ae2
Step 8/22 : RUN python C:\patch-filters-xml.py C:\UnrealEngine\Engine\Build\InstalledEngineFilters.xml && echo. && echo.RUN directive complete. Docker will now commit the filesystem layer to disk. && echo.Note that for large filesystem layers this can take quite some time. && echo.Performing filesystem layer commit... && echo.
 ---> Using cache
 ---> d9b235cdc8a8
Step 9/22 : ARG EXCLUDE_DEBUG
 ---> Using cache
 ---> a50abc80641b
Step 10/22 : ARG EXCLUDE_TEMPLATES
 ---> Using cache
 ---> dccca9f0d92f
Step 11/22 : WORKDIR C:\UnrealEngine
 ---> Using cache
 ---> 61b6ef03342c
Step 12/22 : COPY exclude-components.py C:\exclude-components.py
 ---> Using cache
 ---> ee9f8c7c5404
Step 13/22 : RUN .\Engine\Build\BatchFiles\RunUAT.bat BuildGraph -target="Make Installed Build Win64" -script=Engine/Build/InstalledEngineBuild.xml -set:HostPlatformOnly=true &&       python C:\exclude-components.py C:\UnrealEngine\LocalBuilds\Engine\Windows %EXCLUDE_DEBUG% %EXCLUDE_TEMPLATES% && echo. && echo.RUN directive complete. Docker will now commit the filesystem layer to disk. && echo.Note that for large filesystem layers this can take quite some time. && echo.Performing filesystem layer commit... && echo.
 ---> Using cache
 ---> dbe74ba33173
Step 14/22 : RUN C:\copy.py C:\legacy-toolchain-fix.py C:\UnrealEngine\LocalBuilds\Engine\Windows\ && echo. && echo.RUN directive complete. Docker will now commit the filesystem layer to disk. && echo.Note that for large filesystem layers this can take quite some time. && echo.Performing filesystem layer commit... && echo.
 ---> Using cache
 ---> af9a88661d7c
Step 15/22 : FROM adamrehn/ue4-build-prerequisites:${PREREQS_TAG}
 ---> f1f0f943c9ec
Step 16/22 : COPY --from=builder C:\UnrealEngine\LocalBuilds\Engine\Windows C:\UnrealEngine
failed to copy files: failed to copy directory: write \\?\Volume{e1132e0a-82f1-11e9-ab9d-83e292795455}\UnrealEngine\Engine\Plugins\MovieScene\ActorSequence\Binaries\Win64\UE4Editor-ActorSequence.pdb: There is not enough space on the disk.
[ue4-docker build] Error: failed to build image "adamrehn/ue4-minimal:4.22.1-ltsc2019".
```

**Comment by ki11roy** [comment]
The images produced:
```
PS D:\juyd> docker images
REPOSITORY                               TAG                                IMAGE ID            CREATED             SIZE
<none>                                   <none>                             af9a88661d7c        16 hours ago        196GB
adamrehn/ue4-source                      4.22.1                             e37c1dde37fb        22 hours ago        65.6GB
adamrehn/ue4-source                      4.22.1-ltsc2019                    e37c1dde37fb        22 hours ago        65.6GB
adamrehn/ue4-build-prerequisites         ltsc2019                           f1f0f943c9ec        23 hours ago        23.7GB
<none>                                   <none>                             e541245ddfb6        24 hours ago        9.21GB
mcr.microsoft.com/dotnet/framework/sdk   4.7.2-windowsservercore-ltsc2019   e3f5f2230c2e        9 days ago          7.76GB
```

**Comment by adamrehn** [comment]
This is the exact same error message produced as that seen when [using a version of Docker with the 8GB copy limit bug](https://adamrehn.com/docs/ue4-docker/building-images/troubleshooting-build-issues#building-the-ue4-minimal-image-fails-on-the-copy---frombuilder-directive-that-copies-the-installed-build-from-the-intermediate-image-into-the-final-image), although 18.09.2 should include the bugfix. The solution when working with those older Docker versions was to [exclude debug symbols](https://adamrehn.com/docs/ue4-docker/building-images/advanced-build-options#excluding-engine-components-to-reduce-the-final-image-size), which reduced the overall filesize enough to fall below the 8GB limitation. Out of interest, could you please re-run the build with the flag `--exclude debug` and see if that still results in the same error message?

**Comment by ki11roy** [comment]
Thanks for the fast reply! Sure, will try and report.

**Comment by ki11roy** [comment]
Everything is fine, took quite some time to build though. However the fix dated 2018 it seems that it is still not in 18.09.02 (or there is another similar bug). Thanks!

```
Successfully built 6a04c6ce796d
Successfully tagged adamrehn/ue4-full:4.22.1-ltsc2019
Successfully tagged adamrehn/ue4-full:4.22.1
[ue4-docker build] Built image "adamrehn/ue4-full:4.22.1-ltsc2019" in 20 minutes and 37.2 seconds

[ue4-docker build] Total execution time: 6 hours, 31 minutes and 45.97 seconds
```

**Comment by adamrehn** [comment]
Thanks for testing that. Presumably the fix landed in 18.09.3 then. That's rather frustrating that the latest stable version of Docker for Windows still includes an old version with the bug, since users under Windows Server are already able to update to 18.09.6, which works fine. I've [updated the docs to include the relevant information](https://adamrehn.com/docs/ue4-docker/building-images/troubleshooting-build-issues#building-the-ue4-minimal-image-fails-on-the-copy---frombuilder-directive-that-copies-the-installed-build-from-the-intermediate-image-into-the-final-image) for anyone who might encounter this issue prior to the next release of Docker for Windows.

**Comment by dskvr** [comment]
> 18.09.6, which works fine. I've updated the docs to include the relevant information for anyone who might encounter this issue prior to the next release of Docker for Windows.

@adamrehn Heads up, the documentation lists `18.09.0` not `18.09.6` 

```
The patch that fixes this bug was included in Docker CE 18.09.0, but does not appear to be present in all subsequent builds.
```

**Comment by adamrehn** [comment]
@dskvr that text is still accurate. The bugfix linked by that text was in fact [merged into the upstream source code for Docker CE version 18.09.0](https://github.com/docker/engine/pull/46), but didn't appear to be present in Windows builds until sometime later. Although I know from testing that 18.09.6 seems to include the fix, I don't know for certain which point release was the first that actually contained it under Windows (there's every possibility that it might've been 18.09.3, as suggested above.) The best way to determine whether a given version suffers from the bug is to run [ue4-docker diagnostics 8gig](https://adamrehn.com/docs/ue4-docker/commands/diagnostics#checking-for-the-docker-8gib-filesystem-layer-bug), which is what the documentation now suggests.
